ctapipe/analysis/tests/test_sensitivity.py

```diff
@@ -0,0 +1,219 @@
+    energies_sig = np.ones(Nsig)
+    energies_bgr = np.ones(Nbgr)
+
+    sens = SensitivityPointSource(
+            {'s': energies_sig*u.TeV,   # all events have 1 TeV energy
+             'b': energies_bgr*u.TeV})  # all events have 1 TeV energy
+    sens.event_weights = {'s': energies_sig, 'b': energies_bgr}
+    sensitivity = sens.get_sensitivity(alpha=alpha, signal_list=("s"), mode="MC",
+                                       sensitivity_source_flux=crab_source_rate,
+                                       sensitivity_energy_bin_edges=energy_bin_edges)
+
+    # the ratio between sensitivity and reference flux (i.e. from the crab nebula) should
+    # be the scaling factor that needs to be applied on Nsig to produce a 5 sigma result
+    # in the lima formula
+    ratio = sensitivity["Sensitivity"][0]/(crab_source_rate(1*u.TeV)).value
+    np.testing.assert_allclose([sigma_lima(ratio*Nsig+alpha*Nbgr, Nbgr, alpha)], [5])
+
+
+def test_SensitivityPointSource_sensitivity_data():
+
+    alpha = 1.
+    Non = 30
+    Noff = 10
+    energy_bin_edges = np.array([.1, 10])*u.TeV  # one bin with the bin-centre at 1 TeV
+    energies_sig = np.ones(Non)
+    energies_bgr = np.ones(Noff)
+
+    sens = SensitivityPointSource(
+            {'s': energies_sig*u.TeV,   # all events have 1 TeV energy
+             'b': energies_bgr*u.TeV})  # all events have 1 TeV energy
+
+    sensitivity = sens.get_sensitivity(alpha=alpha, signal_list=("s"), mode="data",
+                                       sensitivity_source_flux=crab_source_rate,
+                                       sensitivity_energy_bin_edges=energy_bin_edges)
+
+    # the ratio between sensitivity and reference flux (i.e. from the crab nebula) should
+    # be the scaling factor that needs to be applied on Nsig to produce a 5 sigma result
+    # in the lima formula
+    ratio = sensitivity["Sensitivity"][0]/(crab_source_rate(1*u.TeV)).value
+    np.testing.assert_allclose([sigma_lima(ratio*(Non-Noff*alpha)+Noff,
+                                           Noff, alpha)], [5])
+
+
+def test_generate_toy_timestamps():
+    from scipy import signal
+    gamma_light_curve = signal.gaussian(5000, 700) * 1000
+
+    tmin, tmax = 0, 5
+
+    Nthrows = 500
+
+    time_stamps = SensitivityPointSource.\
+        generate_toy_timestamps(light_curves={'g': gamma_light_curve,
+                                              'p': Nthrows},
+                                time_window=(tmin, tmax))
+
+    assert len(time_stamps['g']) == int(np.sum(gamma_light_curve))
+    assert len(time_stamps['p']) == Nthrows
+    for st in time_stamps.values():
+        assert (np.min(st) >= tmin) and (np.max(st) <= tmax)
+        np.testing.assert_allclose(np.mean(st), 2.5, atol=.25)
+
+
+def test_draw_events_with_flux_weight():
+
+    Emin = 20
+    Emax = 50
+    Nbin = 50
+    Ndraws = 10000
+
+    def weight_from_energy(e, gamma_new=3, gamma_old=2):
```

Comment:
I don't think that in the current state a function is even necessary... (previously, I wanted to give the callable to the `draw_events_from_flux_weight` method but now it takes the weights-array immediately)
I will put the calculation in-line